apps/test/unit/lib/kits/maker/boards/microBit/MBFirmataUpdaterTest.js

```diff
@@ -0,0 +1,87 @@
+import {expect} from '../../../../../../util/reconfiguredChai';
+import {
+  MICROBIT_V1,
+  MICROBIT_V2,
+  MICROBIT_FIRMATA_V1_URL,
+  MICROBIT_FIRMATA_V2_URL
+} from '@cdo/apps/lib/kits/maker/boards/microBit/MicroBitConstants';
+import MBFirmataUpdater from '@cdo/apps/lib/kits/maker/boards/microBit/MBFirmataUpdater';
+import microBitReducer, {
+  setMicroBitFirmataUpdatePercent
+} from '@cdo/apps/lib/kits/maker/microBitRedux';
+import {
+  getStore,
+  registerReducers,
+  stubRedux,
+  restoreRedux
+} from '@cdo/apps/redux';
+
+describe('MBFirmataUpdater', () => {
+  describe('detectMicroBitVersion function', () => {
+    it('returns correct version for micro:bit v1', () => {
+      const mbFirmataUpdater = new MBFirmataUpdater();
+      const device = {serialNumber: '99001234'};
```

Comment:
Just to be clear, might be good to have a comment here that 9900 is a V1 prefix or just use the constant in the serial number directly (`{serialNumber: MICROBIT_IDS_V1[0] + '1234'}`). Same for V2 below.

For completeness, you could even loop through all ID prefixes, i.e.

```
MICROBIT_IDS_V1.forEach(idPrefix => {
  const device = { serialNumber: idPrefix + '1234'};
  expect ...
}
```

Reply:
Cool idea to loop through all ID prefixes - thanks!